BiweeklyMeetings/2022-09-06.md

```diff
@@ -0,0 +1,353 @@
+
+**Las Safin** -: Or imagine if it's more automated, you can actually do this, even with this native script thing, because what return address do you specify when you send the funds to the docks, not the docks, to the DEX? Well, you want the native script return address, but how do you compute that? It's a hash. Doing that on chain is certainly possible, but it's very inefficient, and I don't see how this solves the problem, myself. I think this is a problem worth solving. I thought about it myself too, but I'm not sure that native scripts are solutions to the problem. I thought maybe what about just adding datums to the address. That's a bit radical, but what are you think about this? How does this solve the problem?
+
+**Sebastien** -: So the key point is that the native script that you're creating can specify a list of scripts that are required, and scripts in Cardano don't have to be deployed on chain, right? So if you want to add extra logic, like how to handle returning something or handling an error case or whatever, you can generate what the Plutus script for that will be, take the hash of it, include that hash inside one of the native script requirements, and then use that as your native script address.
+
+**Las Safin** -: Yeah. I don't think this solves the problem actually, though. I'll give you all a convincing example. Imagine you have some fully automated thing, and I guess the presence of [inaudible 00:30:25] state machine, it does something automatically. At one point, it wants to send funds to a DEX, and the funds should go back to some other script, and the datum will depend on the current state of the state machine. Maybe we just want to send the state machine to the new script actually, and I don't see any way to do this with this mechanism, because there is no way of specifying what datum you want to send back to. Sure, of course, you can specify-
+
+**Sebastien** -: Yes, there is. So basically you create a Plutus script that takes the datum you want to return to as a parameter. Not a runtime parameter, but parameter in the construction of the script itself, like a template.
+
+**Las Safin** -: But that doesn't work, because you can't check this, you can't check this on chain. Although if you check this on chain, you have to do bit-fiddling, you have to check the content of the script, because there's no way to do the parameterisation on chain in a fully automated manner and fully trust this manner on chain, unless you do the bit-fiddling. Of course, that's certainly possible. It doesn't really solve the problem to a satisfactory level in my mind.
```

Comment:
```suggestion
**Las Safin** -: But that doesn't work, because you can't check this, you can't check this on chain. Although if you check this on chain, you have to do bit-fiddling, you have to check the content of the script, because there's no way to do the parameterisation on chain in a fully automated manner and fully trust this manner on chain, unless you do the bit-fiddling. Of course, that's certainly possible. It doesn't really solve the problem to a satisfactory level in my mind.

```